primitives/blockchain/src/header_metadata.rs

```diff
@@ -175,6 +175,12 @@ pub struct TreeRoute<Block: BlockT> {
 	pivot: usize,
 }
 
+impl<Block: BlockT> TreeRoute<Block> {
```

Comment:
This can be stored directly within the other `TreeRoute` impl block just below + add docs.

Furthermore at this point cound be a an idea  to add another (fallible) constructor with the `from` to `params` in place of the stand-alone `tree_route` function.

Reply:
Not sure about adding new `new` wrapping `tree_route` (which requires Backend which is unrelated to `TreeRoute` structure itself.

@davxy how about renaming `new` to `new_test`, as the creation of raw data is used only for tests?

Reply:
[763c51c](https://github.com/paritytech/substrate/pull/12305/commits/763c51c0d0ebda7802f94f75974dcdf43c0ba474)